### ISSUE TITLE
p384: v0.11 release tracking issue

### ISSUE THREAD

**Issue by tarcieri** [RustCrypto/elliptic-curves#566]
This is a tracking ticket for work items it would be nice to have before cutting a final v0.11 release which includes an initial arithmetic implementation.

NOTE: not all of these need to be completed prior to a release and they can be added after-the-fact.

- [x] #573
- [x] Test vectors
  - [x] #567 
  - [x] #570
  - [x] #569
  - [x] #574

cc @brycx @jedisct1

**Comment by tarcieri** [comment]
@daviddrysdale would you be interested in helping add Wycheproof test vectors?

**Comment by tarcieri** [comment]
@jedisct1 did you want to take a look at adding 32-bit support? If not, I probably can in the next few days

**Comment by tarcieri** [comment]
#573 implemented (currently untested) 32-bit support except for field and scalar inversions, which presently use a 64-bit only implementation.

Once that has been addressed, we can add `cross` tests for 32-bit targets, and after that I'd say it's ready to release.

**Comment by daviddrysdale** [comment]
> @daviddrysdale would you be interested in helping add Wycheproof test vectors?

Happy to put in something analogous to the P256 test vectors, but it doesn't seem like the various traits (e.g. `SigningKey`, `VerifyingKey`) needed to use [`new_wycheproof_test!`](https://github.com/RustCrypto/signatures/blob/84768d114c551b799728cbb87e9ae7cef3daac22/ecdsa/src/dev.rs#L137-L221) are available yet?

**Comment by tarcieri** [comment]
@daviddrysdale I think I got it taken care of in #574

**Comment by brycx** [comment]
@tarcieri Is there anything left blocking the `0.11.0` release? I see you've changed the inversion to work with the 32-bit backend as well :rocket:

**Comment by tarcieri** [comment]
#589 adds `const fn` support to field elements

After that it would be nice to expose `const fn` wrappers for the field arithmetic functions, and then rewrite the various constants back into canonical form, transforming them into Montgomery form at compile time.

And after that, I'm good to ship it!

**Comment by tarcieri** [comment]
Released in #593
